src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedValuesFromArray = '';
+
+    for (let index = 0; index < this.length; index++) {
+      if (this[index] !== null && this[index] !== undefined) {
+        joinedValuesFromArray += this[index];
+      }
```

Comment:
I didn't mean you to replace 'i' with 'index'
Developers always use 'i' as variable for iteration

I wanted you to create a variable such as ```const element = this[i] ```

Reply:
roger that